discovery/gossiper.go

```diff
@@ -124,9 +124,12 @@ type Config struct {
 	// should check if we need re-broadcast any of our personal channels.
 	RetransmitDelay time.Duration
 
-	// DB is a global boltdb instance which is needed to pass it in waiting
-	// proof storage to make waiting proofs persistent.
-	DB *channeldb.DB
+	// WaitingProofStore is a persistent storage of partial channel proof
+	// announcement messages. We use it to buffer half of the material
+	// needed to reconstruct a full authenticated channel announcement.
+	// Once we receive the other half the channel proof, we'll be able to
+	// properly validate it and re-broadcast it out to the network.
+	WaitingProofStore *channeldb.WaitingProofStore
```

Comment:
Should add a TOOD here to make this an interface in order to not depend directly on `channeldb` for this functionality.

Reply:
Done.